# HG changeset patch
# User Rik <rik@octave.org>
# Date 1541569110 28800
#      Tue Nov 06 21:38:30 2018 -0800
# Node ID f48504a14fdb587033f0eb8ff999f13b9417f122
# Parent  864448a7f34771c76d555461741e78f66725940b
Stop segfault when starting GUI with -q (bug #54932).

* libgui/qterminal/libqterminal/unix/TerminalView.cpp (mousePressEvent): Check
for lineStart > 0 before indexing with [lineStart-1].

diff --git a/libgui/qterminal/libqterminal/unix/TerminalView.cpp b/libgui/qterminal/libqterminal/unix/TerminalView.cpp
--- a/libgui/qterminal/libqterminal/unix/TerminalView.cpp
+++ b/libgui/qterminal/libqterminal/unix/TerminalView.cpp
@@ -1568,18 +1568,21 @@ void TerminalView::mousePressEvent(QMous
   int charLine;
   int charColumn;
   getCharacterPosition(ev->pos(),charLine,charColumn);
   QPoint pos = QPoint(charColumn,charLine);
 
   // reposition cursor if mouseclick happens in the currently editable line
   QVector<LineProperty> lineprop = _screenWindow->getLineProperties ();
   int lineStart = _screenWindow->cursorPosition ().y ();
-  while (lineprop[lineStart-1] & LINE_WRAPPED)
-    lineStart--;
+  if (lineStart > 0)
+    {
+      while (lineprop[lineStart-1] & LINE_WRAPPED)
+        lineStart--;
+    }
 
   if (charLine >= lineStart)
     {
       int posInLineCursor = _screenWindow->cursorPosition ().x ()
                             + qMax (0,(_screenWindow->cursorPosition ().y ()-lineStart))
                             * _screenWindow->windowColumns ();
       int posInLineClick = charColumn
                            + qMax (0, (charLine - lineStart))
